diff --git a/src/Makefile b/src/Makefile
index 30503b23..dee5d124 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -43,7 +43,6 @@ SRC =              \
 	cortexm.c      \
 	crc32.c        \
 	ch579.c        \
-	efm32.c        \
 	exception.c    \
 	gdb_if.c       \
 	gdb_main.c     \
@@ -97,6 +96,11 @@ SRC =              \
 	target_flash.c \
 	target_probe.c
 
+# on platforms with 128k flash, drop efm32 target to make room
+ifeq (,$(filter f072 native stlink swlink,$(PROBE_HOST)))
+SRC += efm32.c
+endif
+
 ifeq (,$(filter all_platforms,$(MAKECMDGOALS)))
 include $(PLATFORM_DIR)/Makefile.inc
 endif
